lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResult.cs

```diff
@@ -126,6 +139,18 @@ private DriverStatus FetchDriverStatus()
             }
         }
 
+        /// <summary>
+        /// Fetches the Driver Status for the 1st time.
+        /// </summary>
+        /// <returns>The obtained Driver Status or DriverStatus.UNKNOWN, if the Driver was never reached.</returns>
+        private DriverStatus FetchFirstDriverStatus()
+        {
+            var policy = new RetryPolicy<AllErrorsTransientStrategy>(10, TimeSpan.FromMilliseconds(200));
+            DriverStatus result = DriverStatus.UNKNOWN;
+            policy.ExecuteAction(() => result = FetchDriverStatus());
+            return result;
```

Comment:
Can we just write:
```C#
return policy.ExecuteAction<DriverStatus>(FetchDriverStatus);
```
?

Reply:
Also, retry parameters should be probably injectable config parameters with some sane default values, rather than hardcoded constants

Reply:
We can only rewrite as you suggest if we don't need a return value. However, we do.

Reply:
are we talking about the same method? Generic version of `ExecuteAction<>` *[does](https://msdn.microsoft.com/en-us/library/hh675277.aspx)* return a value

Reply:
I made the parameters injectable and also introduced slightly longer defaults.

Reply:
Oh cool! I have updated the code with the generic call.